### DISCUSSION TITLE
Unifi Up Sense

### DISCUSSION THREAD

**Discussion by linkworx-stijnvandenbroucke** [NRCHKB/node-red-contrib-unifi-os#68] [General]
Hi,

The IoT devices are not shown in the 'camera' dropdown but seem to act as a Protect device.
I was verifying how to get for example the temperature from an Up Sense sensor (https://eu.store.ui.com/eu/en/collections/unifi-camera-security-special-sensor/products/up-sense).

Any suggestions how to get data from Unifi IoT devices?

Br, Stijn.

**Comment by crxporter** [comment]
I had one of these a while ago and it was sending data to my nodered ... but I don't have the device anymore.

You can do a call to the bootstrap API (unifi-request node to `/proxy/protect/api/bootstrap`) to grab the current state of everything I think. Grab that endpoint and check under "Sensors" - you should see some data.

![Screenshot 2024-02-04 at 7 47 34 PM](https://github.com/NRCHKB/node-red-contrib-unifi-os/assets/38265886/ff14f56b-3f36-402e-9486-ea76981d867a)

Obviously that's a lot of information. You can either just use it (grab what you need) or you can get the "id" and use that for an HTTP call to the single device. For a camera that unifi-request node will look like a get call to `/proxy/protect/api/cameras/64d6b9a401e...` the word "cameras" in that endpoint might change to "sensors"

Try it out and let me know - I'm sure it is possible but don't have hardware to test.

**Comment by linkworx-stijnvandenbroucke** [comment]
Hi, you are completely right!
Hereunder the json of the first sensor in the array.
Can I coorporate one way or another so there can be 'unifi - protect sensor' building block?
I'm an experienced .net programmer. This languague I haven't used before but I assume it shouldn't be that difficult?

{
	"mac": "D8B3707D654B",
	"host": null,
	"connectionHost": "192.168.10.199",
	"type": "UFP-SENSE",
	"name": "Sensor bureel",
	"upSince": 1707064854769,
	"uptime": null,
	"lastSeen": 1707064842143,
	"connectedSince": 1707064854769,
	"state": "CONNECTED",
	"lastDisconnect": null,
	"hardwareRevision": "113-02823-06",
	"firmwareVersion": "1.2.0",
	"latestFirmwareVersion": "1.2.0",
	"firmwareBuild": null,
	"isUpdating": false,
	"isDownloadingFW": false,
	"fwUpdateState": "upToDate",
	"isAdopting": false,
	"isRestoring": false,
	"isAdopted": true,
	"isAdoptedByOther": false,
	"isProvisioned": false,
	"isRebooting": false,
	"isSshEnabled": false,
	"canAdopt": false,
	"isAttemptingToConnect": false,
	"guid": null,
	"anonymousDeviceId": null,
	"isMotionDetected": false,
	"mountType": "none",
	"leakDetectedAt": null,
	"tamperingDetectedAt": null,
	"isOpened": false,
	"openStatusChangedAt": null,
	"alarmTriggeredAt": null,
	"motionDetectedAt": 1707064116395,
	"wiredConnectionState": {
		"phyRate": null
	},
	"stats": {
		"light": {
			"value": 0,
			"status": "neutral"
		},
		"humidity": {
			"value": 76,
			"status": "neutral"
		},
		"temperature": {
			"value": null,
			"status": "unknown"
		}
	},
	"bluetoothConnectionState": {
		"signalQuality": 90,
		"signalStrength": -54
	},
	"batteryStatus": {
		"percentage": 97,
		"isLow": false
	},
	"alarmSettings": {
		"isEnabled": false
	},
	"lightSettings": {
		"isEnabled": true,
		"lowThreshold": null,
		"highThreshold": null,
		"margin": 10
	},
	"motionSettings": {
		"isEnabled": true,
		"sensitivity": 100
	},
	"temperatureSettings": {
		"isEnabled": false,
		"lowThreshold": null,
		"highThreshold": null,
		"margin": 0.1
	},
	"humiditySettings": {
		"isEnabled": true,
		"lowThreshold": null,
		"highThreshold": null,
		"margin": 1
	},
	"ledSettings": {
		"isEnabled": false
	},
	"bridge": "61a1e46e01ee6703e70003ee",
	"camera": null,
	"bridgeCandidates": [],
	"id": "65ae8c2c0361a603e4050ab6",
	"nvrMac": "74ACB95D1AE6",
	"isConnected": true,
	"marketName": "UP Sense",
	"modelKey": "sensor"
}

**Reply by crxporter** [reply]
At this time I think your best option would be to put an `inject` node to the left of `unifi-request` node. You can set the `inject` to trigger every 5 minutes or 30 seconds or whatever frequency you're looking for to check on the unifi sense data.

---------------
and some notes on push data instead of you grabbing it from the device regularly...

It's very likely this data is included in the websocket alongside the "live" pushed cams data ... but that part of the node is unstable at the moment so I wouldn't recommend it until our next release - which will hopefully be like by the end of March.

Hey @marcus-j-davies what kind of effort would go into adding the sensor device type to the `unifi-protect` node? The "camera" drop-down would need to be called "device" - then just pass all of the messages for that device ID where `camera` is changed to `sensor`? Message me about this whenever you're working on the node again!

edit:
I just remembered these do have motion detectors so a live option would be very beneficial.
